custom_components/tesla_custom/__init__.py

```diff
@@ -504,17 +508,18 @@ def async_update_listeners_debounced(self, delay_since_last=0.1, max_delay=1.0):
 
         """
         # If there's an existing debounce task, cancel it
-        if self._debounce_task:
-            self._debounce_task()
+        if self._cancel_debounce_timer:
+            self._cancel_debounce_timer()
             _LOGGER.debug("Previous debounce task cancelled")
 
-        # Schedule the call to _debounced, pass max_delay using partial
-        self._debounce_task = async_call_later(
-            self.hass, delay_since_last, partial(self._debounced, max_delay)
+        # Schedule the call to _async_debounced, pass max_delay using partial
+        self._cancel_debounce_timer = async_call_later(
```

Comment:
`async_call_later` does not return a task, it returns a callback that can be used to cancel the timer